### PR TITLE
Check whether source Envoy is configured with at least one destination endpoint

### PR DESCRIPTION
**Pull request by draychev**
This PR adds a check - whether the source Envoy has at least one of the Destination endpoints.

Signed-off-by: Delyan Raychev <delyan.raychev@microsoft.com>

### CODE REVIEW

File: pkg/connectivity/pod-to-pod.go [hunk 1]
```diff
@@ -47,6 +47,12 @@ func PodToPod(fromPod *v1.Pod, toPod *v1.Pod) common.Result {
 		namespace.IsMonitoredBy(client, toPod.Namespace, meshName),
 		pod.HasEnvoySidecar(toPod),
 
+		// The source Envoy must have at least one endpoint for the destination Envoy.
+		envoy.HasDestinationEndpoints(srcConfigGetter),
+
+		// Check whether the source Pod has on endpoint, which matches the destination Pod.
```
Comment (shalier) [inline pkg/connectivity/pod-to-pod.go:53]: an endpoint*

File: pkg/envoy/eds.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		return ErrIncorrectlyInitializedConfigGetter
+	}
+	envoyConfig, err := l.ConfigGetter.GetConfig()
+	if err != nil {
+		return err
+	}
+
+	if envoyConfig == nil {
+		return ErrEnvoyConfigEmpty
+	}
```
Comment (shalier) [inline pkg/envoy/eds.go:35]: L33-L35 not needed if L37-L39 exists

File: pkg/envoy/eds.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	envoy_config_endpoint_v3 "github.com/envoyproxy/go-control-plane/envoy/config/endpoint/v3"
+
+	"github.com/openservicemesh/osm-health/pkg/common"
+)
+
+// Verify interface compliance
+var _ common.Runnable = (*HasDestinationEndpointsCheck)(nil)
+
+// HasDestinationEndpointsCheck implements common.Runnable
+type HasDestinationEndpointsCheck struct {
```
Comment (shalier) [inline pkg/envoy/eds.go:17]: Can this be renamed into a noun so it doesn't get confused as a function like L85/L92?
Maybe `DestinationEndpointChecker`?

File: pkg/envoy/eds.go (hunk 2 above)
Comment (shalier) [inline pkg/envoy/eds.go:35]: Not necessary if L37-L39 exists

File: pkg/envoy/eds.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		ConfigGetter: configGetter,
+	}
+}
+
+// HasSpecificEndpoint creates a new common.Runnable, which checks whether the given Pod has an Envoy with properly configured listener for the local payload.
+func HasSpecificEndpoint(configGetter ConfigGetter, pod *v1.Pod) common.Runnable {
+	return HasDestinationEndpointsCheck{
+		ConfigGetter: configGetter,
+		Pod:          pod,
+	}
```
Comment (shalier) [inline pkg/envoy/eds.go:96]: Should these be combined into one function `CreateDestinationEnpointsChecker` that takes in `ConfigGetter` and `*v1.Pod`?

File: pkg/envoy/errors.go [hunk 5]
```diff
@@ -14,4 +14,13 @@ var (
...
+
+	// ErrNoDestinationEndpoints is an error
+	ErrNoDestinationEndpoints = errors.New("no destination endpoints")
+
+	// ErrUnmarshalingClusterLoadAssigment is an error
+	ErrUnmarshalingClusterLoadAssigment = errors.New("error unmarshaling envoy cluster load assigment")
+
+	// ErrEndpointNotFound is an error
+	ErrEndpointNotFound = errors.New("endpoint not found")
 )
```
Comment (shalier) [inline pkg/envoy/errors.go:26]: Can the comments for errors.go be more detailed? Or if it's redundant can we not do the comments?

File: pkg/connectivity/pod-to-pod.go (hunk 1 above)
Comment (draychev) [reply pkg/connectivity/pod-to-pod.go:53]: Thank you!

File: pkg/envoy/eds.go (hunk 4 above)
Comment (draychev) [reply pkg/envoy/eds.go:96]: I want to keep them separate to disambiguate. The one checks for any. The other for specific.

File: pkg/envoy/errors.go (hunk 5 above)
Comment (draychev) [reply pkg/envoy/errors.go:26]: Will augment these!

**Review by johnsonshi — APPROVED** [review]

File: pkg/envoy/eds.go (hunk 2 above)
Comment (draychev) [reply pkg/envoy/eds.go]: This block could indeed be removed!
Thank you @shalier !!!

**Review by shalier — APPROVED** [review]

**Pull request merged**